cosmos/models/Workflow.py

```diff
@@ -443,6 +446,17 @@ def delete(self, delete_files=False):
         self.session.commit()
         print >> sys.stderr, '%s Deleted' % self
 
+    def get_first_failed_task(self, key=lambda t: t.finished_on):
+        """
+        Return the first failed Task (chronologically).
+
+        If no Task failed, return None.
+        """
+        for t in sorted(self.task_graph(), key=key):
```

Comment:
i think if you just filter the Nones out before sorting you'll be good, then you can get rid of the distant future guy.  Also iterating on self.tasks is sufficient, you self.task_graph() constructs a networkx task graph which is overkill